### PR TITLE
[systemtest] Enable few tests that were disabled for KRaft - SCRAM-SHA

### PR DESCRIPTION
**Pull request by im-konge**
### Type of change

- Enhancement

### Description

This PR enables few tests after #8703 - the tests that were disabled contains assumption, which checks if the Kafka version used in the test case is higher or equal to 3.5.0 - as from that version the SCRAM-SHA is working with KRaft.

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 1]
```diff
@@ -403,8 +407,9 @@ void testTlsExternalUser(ExtensionContext extensionContext) {
      *  - secrets
      */
     @ParallelTest
-    @KRaftNotSupported("Probably bug in Kafka - https://issues.apache.org/jira/browse/KAFKA-13964 with fix available in kafka 3.5.0")
     void testUOListeningOnlyUsersInSameCluster(ExtensionContext extensionContext) {
+        assumeTrue(Environment.isKRaftModeEnabled() && TestKafkaVersion.compareDottedVersions("3.5.0", Environment.ST_KAFKA_VERSION) != 1);
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:411]: Should you open some issue to remove all of these once we drop support for Kafka 3.4.x?

**Review by scholzj — APPROVED** [review]

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:411]: Yeah, I wanted to create one, thanks

**Review by Frawless — APPROVED** [review]

**Pull request merged**